api/openapi/api.yaml

```diff
@@ -179,3 +179,146 @@ paths:
           description: Check schematic log
       tags:
         - Model Operations
+  /storage/projects:
+    get:
+      summary: Get all storage projects the current user has access to
+      description: Gets all storage projects the current user has access to, within the scope of the 'storageFileview' attribute.
+      operationId: api.routes.get_storage_projects
+      parameters:
+        - in: query
+          name: input_token
+          schema:
+            type: string
+            nullable: false
+          description: Token
+          example: Token
+          required: true
+        - in: query
+          name: syn_master_file_view
+          schema:
+            type: string
+            nullable: false
+          description: synapse ID of master file view. 
+          example: syn23643253
+          required: true
+        - in: query
+          name: syn_manifest_file_name
+          schema:
+            type: string
+            nullable: false
+          description: Synapse storage manifest file name. 
+          example: synapse_storage_manifest.csv
+          required: true
+      responses:
+        "200":
+          description: A list of tuples
+        "500":
+          description: Check log
+      tags:
+        - Synapse Storage
+  /storage/project/datasets:
+    get:
+      summary: Gets all datasets in folder under a given storage project that the current user has access to.
+      description: Gets all datasets in folder under a given storage project that the current user has access to.
+      operationId: api.routes.get_storage_projects_datasets
+      parameters:
+        - in: query
+          name: input_token
+          schema:
+            type: string
+            nullable: false
+          description: Token
+          example: Token
+          required: true
+        - in: query
+          name: syn_master_file_view
+          schema:
+            type: string
+            nullable: false
+          description: synapse ID of master file view. 
+          example: syn23643253
+          required: true
+        - in: query
+          name: syn_manifest_file_name
```

Comment:
Does the function routed in this endpoint require manifest filename?

Reply:
This is a good point. In SynapseStorage class, I could see: 
`        try:
            self.storageFileview = CONFIG["synapse"]["master_fileview"]
            self.manifest = CONFIG["synapse"]["manifest_filename"]
            self.storageFileviewTable = self.syn.tableQuery(
                    "SELECT * FROM " + self.storageFileview
                ).asDataFrame()
`

That means if we don't add a parameter, our endpoint will load manifest file name from `config.yml`. If that's what we want, I could delete the parameter

Reply:
Note to myself: as we have discussed in the show and tell meeting, we would like to remove `master_fileview` because they are not relevant to the three endpoints that we are adding here.